package.json

```diff
@@ -134,6 +134,7 @@
 		"@hashicorp/react-vertical-text-block-list": "^7.0.0",
 		"@hashicorp/remark-plugins": "^4.0.0",
 		"@hashicorp/sentinel-embedded": "^0.0.14",
+		"@motionone/dom": "^10.14.2",
```

Comment:
Added `@motionone/dom`, this replaces a lot of the complexity of #1091 .

Reply:
Framer motion is already included in the bundle. Is there a benefit to adding motionone here too that we don't get from framer motion?

Reply:
I think so! The perceived benefit is that by using `@motionone/dom`, we can set up a `useScroll` hook that returns `number` values for `{ scrollX, scrollXProgress, scrollY, scrollYProgress }` (where `framer-motion`'s `useScroll` returns `MotionValue<number>`). This gives us flexibility to use `useScroll`:
- We can re-use `ProgressBar`, rather than having to rewrite styles or update that component to use `m.div` & accept `MotionValue<number>` rather than a simpler `number`
- We might be able to re-use `useScroll` for things like `use-scroll-percentage-analytics`
- If we need a `MotionValue`, we can wrap any `number` from `useScroll` in `useMotionValue()`. (Admittedly we can technically go the other way, if `scrollProgressY` was a `MotionValue` I guess we could do `scrollProgressY.get()`, but when I tried this it yielded a value that seemed to update much more infrequently than expected, resulting in laggy-looking scroll progress).

Let me know if that makes sense, happy to adjust and dive into a refactor of `ProgressBar` if there are strong opinions that we should use `framer-motion` `useScroll` directly 🙇

Reply:
Got it, I will leave it up to @BRKalow and team to make the decision on pulling in motionone here vs using Framer Motion.

We seem to be recreating functionality around a library already in use, which feels more brittle in my opinion. Now we're maintaining our own useScroll hook and missing out on some of the benefits you get from motion values. 

- Reduced react rerenders
- subscriptions to values (which seems to be what we'd need to trigger the complete toast on scroll complete?)
- chaining with useTransform hook

> value that seemed to update much more infrequently than expected, resulting in laggy-looking scroll progress

Guessing this by design and is for performance reasons on frequency of updates? This could also be smoothed out with transition easings.

Void of the decision to go the framer or motion one route, this looks good!